X-From-Line: owner-ptk@lists.Stanford.EDU Wed Sep 19 20:20:48 2001
Return-Path: <owner-ptk@lists.Stanford.EDU>
Received: (from eserte@localhost)
	by vran.herceg.de (8.11.4/8.9.3) id f8JIKmE02312
	for eserte; Wed, 19 Sep 2001 20:20:48 +0200 (CEST)
	(envelope-from owner-ptk@lists.Stanford.EDU)
X-Authentication-Warning: vran.herceg.de: eserte set sender to owner-ptk@lists.Stanford.EDU using -f
Received: from mail.cs.tu-berlin.de
	by localhost with POP3 (fetchmail-5.0.5)
	for eserte@localhost (single-drop); Wed, 19 Sep 2001 20:20:47 +0200 (CEST)
Received: from kxmail.berlin.de (kxmail.berlin.de [195.243.105.30])
	by mail.cs.tu-berlin.de (8.9.3/8.9.3) with ESMTP id NAA07456
	for <eserte@cs.tu-berlin.de>; Sat, 8 Sep 2001 13:49:09 +0200 (MET DST)
Received: from lists.Stanford.EDU ([171.64.14.236]) by kxmail.berlin.de
          (InterMail vK.4.03.04.00 201-232-130 license c0e4b842f1eddc5308d584e55543c802)
          with ESMTP
          id <20010908114838.GXAD20038.kxmail@lists.Stanford.EDU>
          for <slaven.rezic@berlin.de>; Sat, 8 Sep 2001 13:48:38 +0200
Received: (from root@localhost)
	by lists.Stanford.EDU (8.11.3/8.11.3) id f889x6n21301
	for ptk-out271740; Sat, 8 Sep 2001 02:59:06 -0700 (PDT)
Received: from mailout04.sul.t-online.de (mailout04.sul.t-online.com [194.25.134.18])
	by lists.Stanford.EDU (8.11.3/8.11.3) with ESMTP id f889x3l21296
	for <ptk@lists.stanford.edu>; Sat, 8 Sep 2001 02:59:04 -0700 (PDT)
Received: from fwd06.sul.t-online.de 
	by mailout04.sul.t-online.de with smtp 
	id 15fet7-000578-02; Sat, 08 Sep 2001 11:58:57 +0200
Received: from familiehaase.dynu.com (320081107336-0001@[193.159.6.110]) by fmrl06.sul.t-online.com
	with smtp id 15fesy-1OebYGC; Sat, 8 Sep 2001 11:58:48 +0200
Received: from loreley.familiehaase.de ([192.168.1.1]) by familiehaase.dynu.com
          with SMTP (Mailtraq/1.1.4.1149) id FMLH1419EC51E;
          Sat, 08 Sep 2001 11:58:33 +0200
Received: by loreley.familiehaase.de (sSMTP sendmail emulation); Sat, 8 Sep 2001 11:58:33 +0200
From: "Gerrit Haase" <gp@familiehaase.de>
Date: Sat, 8 Sep 2001 11:58:33 +0200
X-reported-via: cpan-test version 0.93
Subject: PASS Tk800.023 cygwin-multi 1.3.2(0.3932)
Message-ID: <FMLH1419EC520@familiehaase.dynu.com>
To: cpan-testers@perl.org
Cc: ptk@lists.Stanford.EDU
X-Hops: 1
X-Sender: 320081107336-0001@t-dialin.net
Sender: owner-ptk@lists.Stanford.EDU
Precedence: bulk
X-UIDL: c18eef97ad6b3dc72783ae2b2eccc0db
Lines: 159
Xref: vran.herceg.de pause:13576 ptk:8133

This distribution has been tested as part of the cpan-testers
effort to test as many new uploads to CPAN as possible.  See
http://testers.cpan.org/

Please cc any replies to cpan-testers@perl.org to keep other
test volunteers informed and to prevent any duplicate effort.

-- 
With this patch from Eric Fifer:
================================
diff -urbN Tk800.023-orig/Makefile.PL Tk800.023/Makefile.PL
--- Tk800.023-orig/Makefile.PL	Sat Dec 30 17:12:37 2000
+++ Tk800.023/Makefile.PL	Sat Sep  8 11:47:50 2001
@@ -154,7 +154,7 @@
 }
 
 sub MY::postamble {
-    '
+ my $str = '
 
 html : subdirs manifypods
 	@cd pod && $(MAKE) html $(PASTHRU)
@@ -184,6 +184,8 @@
 	$(MAKE) manifest
 
 ';
+ $str =~ s/DEFINE=.*// if($^O eq 'cygwin');
+ $str;
 }
 
 
diff -urbN Tk800.023-orig/README.cygwin Tk800.023/README.cygwin
--- Tk800.023-orig/README.cygwin	Tue May 15 16:53:37 2001
+++ Tk800.023/README.cygwin	Sat Sep  8 11:52:05 2001
@@ -90,7 +90,7 @@
       (sometimes $^O eq 'MSWin32' is used instead of $Tk::platform
        eq 'MSWin32' assuming they are equivalent and vice versa)
     + #define distinctions
-        _WIN32                defined by gcc, if X11 need to undef
+        _WIN32                win_arch MSWin32
         __WIN32__             win_arch MSWin32, also defined in pTk/Lang.h
         WIN32                 from #include <windows.h>
                               when precedes #include "perl.h" then need
@@ -107,7 +107,7 @@
     README.cygwin MANIFEST
     * documentation
 
-    Makefile.PL Tk/MMutil.pm pTk/Makefile.PL
+    Makefile.PL Tk/MMutil.pm pTk/Makefile.PL myConfig
     * make stuff
 
     pTk/mTk/win/tkWinX.c
@@ -149,6 +149,6 @@
     + Fix gcc warnings, mostly in pTk win32 flavor?
     + Test with cygwin-xfree X Server?
 
-Thu Aug 31 12:54:09 BST 2000
+Sun Jun 10 21:34:30 PST 2001
 Eric Fifer
-efifer@dircon.co.uk
+egf7@columbia.edu
diff -urbN Tk800.023-orig/Tk/MMutil.pm Tk800.023/Tk/MMutil.pm
--- Tk800.023-orig/Tk/MMutil.pm	Sat Dec 30 17:12:37 2000
+++ Tk800.023/Tk/MMutil.pm	Sat Sep  8 11:47:50 2001
@@ -227,13 +227,10 @@
      push(@def,qw(-D__CYGWIN__));
      if ($win_arch eq 'MSWin32')
       {
-       push(@def,qw(-D__WIN32__)) unless $self->{'DEFINE'} =~
/-D__WIN32__/;
+       push(@def,qw(-D__WIN32__ -D_WIN32))
+			       unless $self->{'DEFINE'} =~ /-D__WIN32__/;
        push(@def,qw(-DWIN32)) if $self->{'NAME'} eq 'Tk::pTk';
       }
-     elsif ($win_arch eq 'x')
-      {
-       push(@def,qw(-U_WIN32));
-      }
     }
    foreach (@inc)
     {
@@ -508,15 +505,11 @@
      push(@opt,'LD' => 'gcc -shared');
      if ($win_arch eq 'MSWin32')
       {
-       my $extra = "-lcomdlg32 -lgdi32";
+       my $extra = "-L/lib/w32api -lcomdlg32 -lgdi32";
        my $libs = $att{'LIBS'}->[0];
        $att{'LIBS'}->[0] = "$extra $libs";
-       $att{'DEFINE'} .= ' -D__WIN32__';
+       $att{'DEFINE'} .= ' -D__WIN32__ -D_WIN32';
        $att{'DEFINE'} .= ' -DWIN32' if($att{'NAME'} eq 'Tk::pTk');
-      }
-     elsif ($win_arch eq 'x')
-      {
-       $att{'DEFINE'} .= ' -U_WIN32';
       }
     }
    if (delete $att{'ptk_include'})
diff -urbN Tk800.023-orig/myConfig Tk800.023/myConfig
--- Tk800.023-orig/myConfig	Sat Dec 30 17:12:37 2000
+++ Tk800.023/myConfig	Sat Sep  8 11:47:50 2001
@@ -340,7 +340,7 @@
   $define .= " -D__PM__";
   $inc = "-I../pTk/mTk/xlib -I../../pTk/mTk/xlib -IpTk/mTk/xlib
-ImTk/os2_rc";
   $xlib = "";                          # No library is needed
-} elsif ($IsWin32) {
+} elsif ($IsWin32 or $^O eq 'cygwin') {
   $inc = '-I$(TKDIR)/pTk/mTk/xlib';
 }
 
diff -urbN Tk800.023-orig/pTk/Xlib.excwin Tk800.023/pTk/Xlib.excwin
--- Tk800.023-orig/pTk/Xlib.excwin	Sat Dec 30 17:12:37 2000
+++ Tk800.023/pTk/Xlib.excwin	Sat Sep  8 11:47:50 2001
@@ -38,3 +38,4 @@
 XOpenDisplay
 XLoadQueryFont
 XCreateWindow
\ No newline at end of file
+XRectInRegion

=====================================END OF PATCH==================
--

Summary of my perl5 (revision 5.0 version 6 subversion 1) configuration:
  Platform:
    osname=cygwin, osvers=1.3.2(0.3932), archname=cygwin-multi
    uname='cygwin_nt-4.0 loreley 1.3.2(0.3932) 2001-05-20 23:28 i686 unknown '
    config_args='-de -Dusemultiplicity'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=define
    useperlio=undef d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
  Compiler:
    cc='gcc', ccflags ='-DPERL_USE_SAFE_PUTENV -fno-strict-aliasing -I/usr/local/include',
    optimize='-O2',
    cppflags='-DPERL_USE_SAFE_PUTENV -fno-strict-aliasing -I/usr/local/include'
    ccversion='', gccversion='2.95.3-5 (cygwin special)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=8, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='ld2', ldflags =' -s -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib /lib
    libs=-lgdbm -lcrypt
    perllibs=-lcrypt
    libc=/usr/lib/libc.a, so=dll, useshrplib=true, libperl=libperl5_6_1.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' -s'
    cccdlflags=' ', lddlflags=' -s -L/usr/local/lib'

-++**==--++**==--++**==--++**==--++**==--++**==--++**==
This message was posted through the Stanford campus mailing list
server.  If you wish to unsubscribe from this mailing list, send the
message body of "unsubscribe ptk" to majordomo@lists.stanford.edu


